form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowStepInstance.kt

```diff
@@ -45,6 +47,40 @@ data class FormFlowStepInstance(
     val definition: FormFlowStep
         get() = instance.formFlowDefinition.getStepByKey(stepKey)
 
+    fun open() {
+        processExpressions(FormFlowStep::onOpen)
+    }
+
+    fun complete(submissionData: String) {
+        this.submissionData = submissionData
+
+        processExpressions(FormFlowStep::onComplete)
+    }
+
+    private fun processExpressions(expressionist: (FormFlowStep)-> List<String>?) {
+        ExpressionProcessorFactoryHolder.getinstance()?.let {
```

Comment:
The ExpressionProcessorFactoryHolder is one of many ways to call a bean from a domain object. 

In my opinion, you should never call a bean from a domain object. But it seems no one agree with me. So if we really want to do that, we should at least sit together and decide on a single way to call beans from a domain class. Because I've seen several ways now